test/powershell/engine/Encoding.Tests.ps1

```diff
@@ -0,0 +1,173 @@
+
+        $contentTests =
+            @{ Name = 'Ascii'; Bytes = "116-63-115-116-" + (Get-NewLineBytes Ascii) },
+            @{ Name = 'BigEndianUTF32'; Bytes = "254-255-0-116-0-233-0-115-0-116-" + (Get-NewLineBytes BigEndianUTF32) },
+            @{ Name = 'BigEndianUnicode'; Bytes = "254-255-0-116-0-233-0-115-0-116-" + (Get-NewLineBytes BigEndianUnicode) },
+            @{ Name = 'Byte'; Bytes = "255-254-116-0-233-0-115-0-116-0-" + (Get-NewLineBytes Byte) },
+            @{ Name = 'Default'; Bytes = "116-195-169-115-116-" + (Get-NewLineBytes Default) },
+            # Oem encoding can change depending on system, calculate the expected string
+            @{ Name = 'Oem'; Bytes = ([Microsoft.PowerShell.PowerShellEncoding]::GetEncoding("Oem").GetBytes($testString) -join "-") + "-" + (Get-NewLineBytes Oem) },
+            @{ Name = 'String'; Bytes = "255-254-116-0-233-0-115-0-116-0-" + (Get-NewLineBytes String) },
+            @{ Name = 'UTF32'; Bytes = "255-254-0-0-116-0-0-0-233-0-0-0-115-0-0-0-116-0-0-0-" + (Get-NewLineBytes UTF32) },
+            @{ Name = 'UTF7'; Bytes = "116-43-65-79-107-45-115-116-" + (Get-NewLineBytes UTF7) },
+            @{ Name = 'UTF8'; Bytes = "116-195-169-115-116-" + (Get-NewLineBytes UTF8 ) },
+            @{ Name = 'UTF8BOM'; Bytes = "239-187-191-116-195-169-115-116-" + (Get-NewLineBytes UTF8BOM) },
+            @{ Name = 'UTF8NoBOM'; Bytes = "116-195-169-115-116-" + (Get-NewLineBytes UTF8NoBOM) },
+            @{ Name = 'Unicode'; Bytes = "255-254-116-0-233-0-115-0-116-0-" + (Get-NewLineBytes Unicode) },
+            @{ Name = 'Unknown'; Bytes = "116-195-169-115-116-" + (Get-NewLineBytes Unknown) }
+
+    }
+
+    AfterEach {
+        if ( Test-Path $testFile )
+        {
+            remove-item $testFile
+        }
+        $PSDefaultFileEncoding = "Unknown"
+    }
+
+    It "Encoding for '<Name>' should have correct preamble '<preamble>'" -TestCase $preambleTests {
+        param ( $Name, $Preamble )
+        [Microsoft.PowerShell.PowerShellEncoding]::GetEncoding($Name).GetPreamble() -Join "-" | Should be $Preamble
+    }
+
+    It "Encoding for '<Name>' should create file with proper encoding" -TestCase $contentTests {
+        param ( $Name, $Bytes )
+        $testString | out-file -encoding $Name $testFile
+        Get-FileBytes $testFile | should be $Bytes
+    }
+
+    It "Setting PSDefaultFileEncoding to '<Name>' should create file with proper encoding" -TestCase $contentTests {
+        param ( $Name, $Bytes )
+        $PSDefaultFileEncoding = $Name
+        $testString | out-file $testFile
+        Get-FileBytes $testFile | should be $Bytes
+    }
+
+    It "Explicit encoding is not overridden by setting PSDefaultFileEncoding to '<Name>'" -TestCase $contentTests {
+        param ( $Name, $Bytes )
+        $PSDefaultFileEncoding = $Name
+        $testString | out-file -encoding ascii $testFile
+        Get-FileBytes $testFile | should be ("116-63-115-116-" + (Get-NewLineBytes ASCII))
```

Comment:
A brief comment explaining why this is handled specially with the leading bytes spelled out would be useful for this test and subsequent tests.